x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/transforms/TransformConfig.java

```diff
@@ -176,11 +183,12 @@ public static String documentId(String transformId) {
         this.syncConfig = syncConfig;
         this.setHeaders(headers == null ? Collections.emptyMap() : headers);
         this.pivotConfig = pivotConfig;
+        this.latestDocConfig = latestDocConfig;
         this.description = description;
         this.settings = settings == null ? new SettingsConfig() : settings;
 
-        // at least one function must be defined
-        if (this.pivotConfig == null) {
+        // exactly one function must be defined
+        if ((this.pivotConfig == null) == (this.latestDocConfig == null)) {
             throw new IllegalArgumentException(TransformMessages.TRANSFORM_CONFIGURATION_NO_TRANSFORM);
```

Comment:
`TRANSFORM_CONFIGURATION_NO_TRANSFORM` could you rename this message constant? I know the message is probably OK (since it says "exactly 1") but the constant name implies something else.

Reply:
Done (changed to `TRANSFORM_CONFIGURATION_BAD_FUNCTION_COUNT`).